src/engine/shared/map.cpp

```diff
@@ -55,7 +56,14 @@ class CMap : public IEngineMap
 					
 					if(pTilemap->m_Version > 3)
 					{
+						if(!((pTilemap->m_Width <= INT_MAX / 2 - 1) && (pTilemap->m_Height <= INT_MAX / 2 - 1)))
```

Comment:
You can still have an integer overflow. F.e.: (INT_MAX/2-2)*(INT_MAX/2-2) > INT_MAX

Reply:
ugh, I took the wrong half of #2074